tensorflow_datasets/scripts/replace_fake_images.py

```diff
@@ -52,7 +52,7 @@ def rewrite_image(filepath):
   image = np.array(PIL.Image.open(filepath))
 
   # Filter unsuported images
-  if image.mode == 'RGBA' or image.dtype == np.bool:
+  if PIL.Image.open(filepath).mode == 'RGBA' or image.dtype == np.bool:
```

Comment:
You should re-use the pil image open above, rather than re-opening the image twice.